core/types/block.go

```diff
@@ -252,12 +252,16 @@ type extblock struct {
 	Requests    []*Request    `rlp:"optional"`
 }
 
+type BlockType interface {
+	HasOptimismWithdrawalsRoot(blkTime uint64) bool
```

Comment:
Renamed `HasWithdrawalsRoot` -> `HasOptimismWithdrawalsRoot` per suggestion from @sebastianst and @protolambda